### PR TITLE
Do not override my config

### PR DESCRIPTION
**Pull request by staxmanade**
# TL;DR

`react-wildcat` should never override a value the developer specifies in `wildcat.config.js`.

I have 1 commit here
  - [x] the actual fix (switching the merge ordering) and some tests

# The long version

Trying to work through Heroku deployment. I got to the point where my whole site could be run through a single server.

The last problem was trying to get the template rendered to use my custom domain to render the root template with `http://mysite.herokuapp.com/jspm_packages/system.js`, however this always rendered with `http://localhost:12345/jspm_packages/system.js`. 

I tried updating the `wildcatConfig.serverSettings.staticServer.hostname/port` but each time I touched those values I only ever got `500` responses from heroku (with no error logs that I could find).

After digging in, I found out about the `staticUrl` and `originUrl` values. When I tried to hard-code them in my `wildcat.config.js` that still didn't seem to work. 

Debugging further I found out about the merging of configs happening inside `getWildcatConfig.js` which looked to have the `merge` with the convenience properties `originUrl` & `staticUrl` added on.

I prototyped a hack-fix (without touching wildcat the project) by using the following:

```
var Module             = require('module')
    , originalLoader   = Module._load
Module._load = function(request, parent) {
    if(request && request.indexOf('getWildcatConfig') >= 0) {
        var originalGetWildcatConfig =  originalLoader.apply(this, arguments);
        return function newGetWildcatConfig(cfg) {
            var config = originalGetWildcatConfig(cfg);
            // force-override
            config.generalSettings.originUrl = "http://mysite.herokuapp.com";
            config.generalSettings.staticUrl = "http://mysite.herokuapp.com";
            console.log("re-hacking config");
            console.log(config);
            return config;
        }
    }
    return originalLoader.apply(this, arguments);
};
```

and :boom: I was in business...

So, while I don't know why I couldn't just change the config static `hostname: "localhost"` to `hostname: "mysite.herokuapp.com"`, I still don't think that the framework should be overriding anything I manually stick in the `wildcat.config.js`.

/cc @doctyper

### CODE REVIEW

File: packages/react-wildcat/test/getWildcatConfigSpec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    context("when app config HAS specified values config values", () => {
+        let wildcatConfig;
+
+        before(() => {
+            var projectConfigFile = "./wildcat.config.js";
+            console.log("XXX\n\n\n\n" + projectConfigFile + "\n\n\n");
+
+            const CustomGetWildcatConfig = proxyquire("../src/utils/getWildcatConfig.js", {
+                '???STUCK_HERE???/wildcat.config.js': () => {
```
Comment (doctyper) [inline packages/react-wildcat/test/getWildcatConfigSpec.js]: try `${exampleDir}/wildcat.config.js`

**Comment by staxmanade** [comment]
@doctyper Worked through my test setup issues... Let me know how this looks.

**Comment by doctyper** [comment]
Looks great, thanks!
